app/shop.py

```diff
@@ -0,0 +1,34 @@
+import datetime
+
+
+class Shop:
+    def __init__(
+            self,
+            name: str,
+            location: list[int],
+            products: dict
+    ) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+        self.each_product_cost = {}
+        self.total_cost = 0
+
+    def purchase(self, product_cart: dict) -> int:
+        for product in product_cart:
+            product_cost = product_cart[product] * self.products[product]
+            if product_cost.is_integer():
+                product_cost = int(product_cost)
+            self.each_product_cost[product] = product_cost
+            self.total_cost += product_cost
+        return self.total_cost
+
+    def receipt(self, customer_name: str, product_cart: dict) -> None:
+        current_time = datetime.datetime.now().strftime("%d/%m/%Y %H:%M:%S")
+        print(f"Date: {current_time}\nThanks, "
+              f"{customer_name}, for your purchase!")
+        print("You have bought:")
+        for product in self.each_product_cost:
+            print(f"{product_cart[product]} {product}s "
+                  f"for {self.each_product_cost[product]} dollars")

```

Comment:
This loop assumes that every product in `self.each_product_cost` is also in `product_cart`. If there are discrepancies between these dictionaries, it could lead to errors. Ensure that the dictionaries are synchronized or handle cases where they might differ.